activitysim/core/assign.py

```diff
@@ -183,9 +184,11 @@ def assign_variables(assignment_expressions, df, locals_dict, df_alias=None,
     lowercase variables starting with underscore are temp variables (e.g. _local_var)
     and not returned except in trace_results
 
-    uppercase variables starting with underscore are temp scalar variables (e.g. _LOCAL_SCALAR)
+    uppercase variables starting with underscore are temp singular variables (e.g. _LOCAL_SCALAR)
```

Comment:
What is the reasoning behind changing the nomenclature?

Reply:
In translating CTRAMP UEC's to ActivitySim expressions, I encountered a number of instances where it was useful to be able to create and store complex objects (e.g. numpy matrices, ordered dictionaries, etc.) that do not vary across choosers or alternatives. I wanted to piggyback on the functionality of the _LOCAL_SCALAR syntax, but in doing so discovered that it already works for objects more complex than scalars. I changed the docstrings and internal method names to better reflect this capability.